core/runtime/src/main/java/io/quarkiverse/mcp/server/Cancellations.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkiverse.mcp.server;
+
+/**
+ * Service for tracking cancelled MCP requests
+ */
+public interface Cancellations {
+
+    /**
+     * Check if a request has been cancelled (MCP compliant)
+     *
+     * @param requestId the request identifier from MCP protocol
+     * @return true if the request has been cancelled, false otherwise
+     */
+    boolean isCancelled(RequestId requestId);
+
+    /**
+     * Check if a request has been cancelled for a specific connection
+     *
+     * @param connectionId the connection identifier
+     * @param requestId the request identifier
+     * @return true if the request has been cancelled, false otherwise
+     */
+    boolean isCancelled(String connectionId, RequestId requestId);
```

Comment:
While looking at this API, I wonder if from user POV it wouldn't we easier to consume an interface like:

```java
interface Cancellation {
   boolean isRequested();
}
```

direclty in a feature method, i.e. something like:

```java
class Tools {

   String foo(Cancellation cancellation) {
      // do something
      if (cancellation.isRequested()) {
         throw new OperationCancelledException();
      }
   }

}
```
This way a user would not need to pass neither the request id nor the connection id :thinking:.

Reply:
Ok, it would be great if you could take a look at https://github.com/quarkiverse/quarkus-mcp-server/pull/332.